lib/backend/report.go

```diff
@@ -173,6 +173,8 @@ func (s *Reporter) Create(ctx context.Context, i Item) (*Lease, error) {
 	lease, err := s.Backend.Create(ctx, i)
 	writeLatencies.WithLabelValues(s.Component).Observe(s.Clock().Since(start).Seconds())
 	writeRequests.WithLabelValues(s.Component).Inc()
+	writeRequestsFailed.WithLabelValues(s.Component)
+	writeRequestsFailedPrecondition.WithLabelValues(s.Component)
```

Comment:
These initializations are still only going to happen on the first Create - should they be in `NewReporter` instead? We might also benefit from storing the metrics with the curried labels in the struct rather than calling WithLabelValues on the global value every time.

Reply:
> These initializations are still only going to happen on the first Create - should they be in NewReporter instead?

I had wondered this too.  The Create (write) and Get (read) happen enough for it to work but feels off and is subject to edge cases.   I'd be open to moving them to `NewReporter`.

Reply:
Let me know how [f756a16](https://github.com/gravitational/teleport/pull/50092/commits/f756a16795d015f7f5ee52b30ae980e09c6679bf) looks